chalk-parse/src/parser.lalrpop

```diff
@@ -188,7 +189,33 @@ TyWithoutFor: Ty = {
     },
     <n:Id> "<" <a:Comma<Parameter>> ">" => Ty::Apply { name: n, args: a },
     <a:AliasTy> => Ty::Alias { alias: a },
-    "(" <Ty> ")",
+    "(" <t:TupleOrParensInner> ")" => t,
+};
+
+ScalarType: ScalarType = {
+   "u8" => ScalarType::Uint(UintTy::U8),
+   "u16" => ScalarType::Uint(UintTy::U16),
+   "u32" => ScalarType::Uint(UintTy::U32),
+   "u64" => ScalarType::Uint(UintTy::U64),
+   "u128" => ScalarType::Uint(UintTy::U128),
+   "usize" => ScalarType::Uint(UintTy::Usize),
+   "i8" => ScalarType::Int(IntTy::I8),
+   "i16" => ScalarType::Int(IntTy::I16),
+   "i32" => ScalarType::Int(IntTy::I32),
+   "i64" => ScalarType::Int(IntTy::I64),
+   "i128" => ScalarType::Int(IntTy::I128),
+   "isize" => ScalarType::Int(IntTy::Isize),
+   "f32" => ScalarType::Float(FloatTy::F32),
+   "f64" => ScalarType::Float(FloatTy::F64),
+   "bool" => ScalarType::Bool,
+   "char" => ScalarType::Char,
+};
+
+TupleOrParensInner: Ty = {
+    <l:Comma2<Ty>> => Ty::Tuple { arity: l.len() }, 
```

Comment:
Mm yeah it'd nice to not have `Comma2`.

Try this:

```
TupleOrParensInner: Ty = {
    <Ty>,
    <Ty> "," => Ty::Tuple { types: vec[<>] },
    <a:Ty> "," <types:Comma<Ty>>  => {
        let mut types  = types;
        types.insert(0, a);
        Ty::Tuple { types }
    }
```

(we can always extend `Comma` to accept trailing commas, too, but that's separate)

Reply:
not mega efficient but who cares